tools/scenario-player/scenario_player/tasks/blockchain.py

```diff
@@ -103,12 +104,27 @@ def __init__(
                 'Not all required keys provided. Required: ' + ', '.join(required_keys),
             )
 
-        self.contract_name = config.get('contract_name', None)
-        self.event_name = config.get('event_name', None)
-        self.num_events = config.get('num_events', 0)
+        self.contract_name = config.get('contract_name')
+        self.event_name = config.get('event_name')
+        self.num_events = config.get('num_events')
 
         self.web3 = self._runner.client.web3
 
+        # get the correct contract address
+        service_contract_data = get_contracts_deployed(
+            chain_id=self._runner.chain_id,
+            version=None,
```

Comment:
Not sure if we should bind this to the raiden contracts version.

Reply:
We should... you can use `DEVELOPMENT_CONTRACT_VERSION` from `raiden.settings`

Reply:
But then we limit the SP, maybe a setting for the scenario with the raiden version being the default makes most sense

Reply:
hmm, if we set Raiden version to XXX and use `latest` contracts then there might be some differences between what the client expects vs what the contracts provide. I don't think it should be a problem right now so this is ok for me, but i am just commenting on the potential of future issues.

Reply:
Changed to raiden version, will add setting later